boards/arc/emsdp/emsdp-pinctrl.dtsi

```diff
@@ -0,0 +1,124 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <zephyr/dt-bindings/pinctrl/emsdp-pinctrl.h>
+
+&pinctrl {
+	/* PMOD_A */
+	pmodA_gpio: pmodA_gpio {
+		pinmux = <PMOD_A PMOD_GPIO>;
+	};
+	pmodA_uart: pmodA_uart {
+		pinmux = <PMOD_A PMOD_UART>;
+	};
+	pmodA_spi1_cs0: pmodA_spi1_cs0 {
+		pinmux = <PMOD_A PMOD_SPI>;
+	};
+	pmodA_i2c2: pmodA_i2c2 {
+		pinmux = <PMOD_A PMOD_I2C>;
+	};
+	pmodA_pwm1: pmodA_pwm1 {
+		pinmux = <PMOD_A PMOD_PWM_MODE1>;
+	};
+	pmodA_pwm2: pmodA_pwm2 {
+		pinmux = <PMOD_A PMOD_PWM_MODE2>;
+	};
+
+	/* PMOD_B */
+	pmodB_gpio: pmodB_gpio {
+		pinmux = <PMOD_B PMOD_GPIO>;
+	};
+	pmodB_uart: pmodB_uart {
+		pinmux = <PMOD_B PMOD_UART>;
+	};
+	pmodB_spi1_cs1: pmodB_spi1_cs1 {
+		pinmux = <PMOD_B PMOD_SPI>;
+	};
+	pmodB_i2c2: pmodB_i2c2 {
+		pinmux = <PMOD_B PMOD_I2C>;
+	};
+	pmodB_pwm1: pmodB_pwm1 {
+		pinmux = <PMOD_B PMOD_PWM_MODE1>;
+	};
+	pmodB_pwm2: pmodB_pwm2 {
+		pinmux = <PMOD_B PMOD_PWM_MODE2>;
+	};
+
+	/* PMOD_C */
+	pmodC_gpio: pmodC_gpio {
+		pinmux = <PMOD_C PMOD_GPIO>;
+	};
+	pmodC_uart: pmodC_uart {
+		pinmux = <PMOD_C PMOD_UART>;
+	};
+	pmodC_spi1_cs2: pmodC_spi1_cs2 {
+		pinmux = <PMOD_C PMOD_SPI>;
+	};
+	pmodC_i2c2: pmodC_i2c2 {
+		pinmux = <PMOD_C PMOD_I2C>;
+	};
+	pmodC_pwm1: pmodC_pwm1 {
+		pinmux = <PMOD_C PMOD_PWM_MODE1>;
+	};
+	pmodC_pwm2: pmodC_pwm2 {
+		pinmux = <PMOD_C PMOD_PWM_MODE2>;
+	};
+
+	/* ARDUINO_PIN_1 */
+	arduino1_gpio: arduino1_gpio {
```

Comment:
The pinmux has some constraints, not all pins can be individually set. I looks like it was decided to only explicitly list the highest number pin for each group that is programmed together?
That would at least require some explanation (documentation is lacking), or better use the name of the group instead of the individual pins (e.g. arduino_IO0_IO1).

Reply:
updated in doc and renamed to arduino_CFG_xxx